airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_stream.py

```diff
@@ -194,7 +202,7 @@ async def _infer_file_schema(self, file: RemoteFile) -> Mapping[str, Any]:
             )
```

Comment:
For my curiosity (and therefore outside of the PR scope): why is infer_schema async?